torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2595,6 +2595,54 @@ Tensor linalg_qr_backward(const std::vector<torch::autograd::Variable> &grads, c
   }
 }
 
+// The backward for this function is just a specialized version of
+// lu.backward, which is implemented in /torch/_autograd_functions.py
+Tensor _det_lu_based_helper_backward(
+  const Tensor& det_grad,
+  const Tensor& det,
+  const Tensor& self,
+  const Tensor& p,
+  const Tensor& l,
+  const Tensor& u
+) {
+  if (!det_grad.defined()) {
+    return at::zeros_like(self, at::MemoryFormat::Contiguous);
+  }
+
+  auto eps = at::native::_get_epsilon(c10::toValueType(self.scalar_type()));
+  auto n = self.size(-1);
+
+  auto u_h = u.transpose(-2, -1).conj();
+  auto u_h_diag = u_h.diagonal(0, -2, -1);
+  auto u_h_diag_conditioned = at::where(
+    u_h_diag == 0.0,
+    at::tensor(eps, self.options()),
+    u_h_diag
+  );
+  u_h_diag.copy_(u_h_diag_conditioned);
+
+  auto l_h = l.transpose(-2, -1).conj();
+
+  // create a matrix d := det_grad * det.conj() * I
+  auto det_expanded_sizes = det.sizes().vec();
+  det_expanded_sizes.push_back(n);
+  auto d = at::diag_embed((det_grad * det.conj()).unsqueeze(-1).expand(det_expanded_sizes));
+
+  // permuted_grad := l_h^{-1} d u_h^{-1}, similar to lu.backward
+  auto permuted_grad = std::get<0>(
+    at::triangular_solve(
+      // note that d = c I for some scalar c, hence
+      // d u_h^{-1} = c I u_h^{-1} = u_h^{-1} c I = u_h^{-1} d,
+      // so, there is no need to explicitly transpose the solution below
+      std::get<0>(at::triangular_solve(d, u_h, /*upper=*/false)),
+      l_h
```

Comment:
```suggestion
    at::triangular_solve(
      // note that d = c I for some scalar c, hence
      // d u_h^{-1} = c I u_h^{-1} = u_h^{-1} c I = u_h^{-1} d,
      // so, there is no need to explicitly transpose the solution below
      std::get<0>(at::triangular_solve(d, u_h, /*upper=*/false)),
      l_h, /*upper=*/false, /*transpose=*/false, /*unitriangular=*/true
```
breaks the code! Which makes me wonder whether this parameter works as expected. Related to other issues pointed out with `triangular_solve`.